From 81ae1fab6c21ec7bb78766e81d9eb3042376cc87 Mon Sep 17 00:00:00 2001
From: Vitaly Prosyak <vitaly.prosyak@amd.com>
Date: Thu, 14 Jul 2016 16:55:55 -0400
Subject: [PATCH 0618/1722] drm/amdgpu: add flip flags in amdgpu_crtc struct

Temporally add to amdgpu_crtc flip_flags because this flag is not supported by
atomic framework.The alternative and probably a proper solution would be decouple
page flip from atomic, but it may take a while because synchronization is required.
The current dal atomic code relies on atomic locks.Async flip would be important
for free sync, therefore i am proposing this quick fix.

Change-Id: I706bf1ff3c8df60faee03e1129fa52fb658c9728
Signed-off-by: Vitaly Prosyak <vitaly.prosyak@amd.com>
Signed-off-by: Kalyan Alle <kalyan.alle@amd.com>
---
 drivers/gpu/drm/amd/amdgpu/amdgpu_mode.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu_mode.h b/drivers/gpu/drm/amd/amdgpu/amdgpu_mode.h
index 502a20b..a2f12b0 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_mode.h
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_mode.h
@@ -438,8 +438,9 @@ struct amdgpu_crtc {
 	u32 lb_vblank_lead_lines;
 	struct drm_display_mode hw_mode;
 
-	/* After Set Mode target will be non-NULL */
 	int otg_inst;
+	uint32_t flip_flags;
+	/* After Set Mode target will be non-NULL */
 	struct dc_target *target;
 };
 
-- 
2.7.4

